Makefile

```diff
@@ -0,0 +1,201 @@
+
+.PHONY: lint
+lint:
+	$(GO_DOCKER_CMD) golangci-lint run --config golangci.yaml
+
+.PHONY: test-unit
+test-unit:
+	$(GO_DOCKER_CMD) go test \
+		-v \
+		-timeout=60s \
+		-race \
+		-coverprofile=coverage.txt \
+		-covermode=atomic \
+		./...
+
+.PHONY: lint-chart
+lint-chart:
+	$(HELM_DOCKER_CMD) sh -c ' \
+		cd charts/brigade-slack-gateway && \
+		helm dep up && \
+		helm lint . \
+	'
+
+################################################################################
+# Build / Publish                                                              #
+################################################################################
+
+.PHONY: build
+build: build-images
+
+.PHONY: build-images
+build-images: build-receiver build-monitor
+
+.PHONY: build-%
+build-%:
+	$(KANIKO_DOCKER_CMD) kaniko \
+		--build-arg VERSION=$(VERSION) \
+		--build-arg COMMIT=$(GIT_VERSION) \
+		--dockerfile /workspaces/brigade-slack-gateway/$*/Dockerfile \
+		--context dir:///workspaces/brigade-slack-gateway/ \
+		--no-push
+
+################################################################################
+# Publish                                                                      #
+################################################################################
+
+.PHONY: publish
+publish: push-images publish-chart
+
+.PHONY: push-images
+push-images: push-receiver push-monitor
+
+.PHONY: push-%
+push-%:
+	$(KANIKO_DOCKER_CMD) sh -c ' \
+		docker login $(DOCKER_REGISTRY) -u $(DOCKER_USERNAME) -p $${DOCKER_PASSWORD} && \
+		kaniko \
+			--build-arg VERSION="$(VERSION)" \
+			--build-arg COMMIT="$(GIT_VERSION)" \
+			--dockerfile /workspaces/brigade-slack-gateway/$*/Dockerfile \
+			--context dir:///workspaces/brigade-slack-gateway/ \
+			--destination $(DOCKER_IMAGE_PREFIX)$*:$(IMMUTABLE_DOCKER_TAG) \
+			--destination $(DOCKER_IMAGE_PREFIX)$*:$(MUTABLE_DOCKER_TAG) \
+	'
+
+.PHONY: publish-chart
+publish-chart:
+	$(HELM_DOCKER_CMD) sh	-c ' \
+		helm registry login $(HELM_REGISTRY) -u $(HELM_USERNAME) -p $${HELM_PASSWORD} && \
+		cd charts/brigade-slack-gateway && \
+		helm dep up && \
+		helm package . --version $(VERSION) --app-version $(VERSION) && \
+		helm push brigade-slack-gateway-$(VERSION).tgz oci://$(HELM_REGISTRY)$(HELM_ORG) \
+	'
+
+################################################################################
+# Targets to facilitate hacking on this gateway.                               #
+################################################################################
+
+.PHONY: hack-build-%
+hack-build-%:
+	docker build \
+		-f $*/Dockerfile \
+		-t $(DOCKER_IMAGE_PREFIX)$*:$(IMMUTABLE_DOCKER_TAG) \
+		--build-arg VERSION='$(VERSION)' \
+		--build-arg COMMIT='$(GIT_VERSION)' \
+		.
+
+.PHONY: hack-push-images
+hack-push-images: hack-push-receiver hack-push-monitor
+
+.PHONY: hack-push-%
+hack-push-%: hack-build-%
+	docker push $(DOCKER_IMAGE_PREFIX)$*:$(IMMUTABLE_DOCKER_TAG)
+
+IMAGE_PULL_POLICY ?= Always
+
+.PHONY: hack-deploy
+hack-deploy:
```

Comment:
To deploy successfully, a value for `brigade.apiToken` must be provided.  We could supply a bogus value here just so the helm deploy is successful, but of course the monitor pod would crash loop.  We could update this target to take a filepath to a values.yaml file, presumably with the `brigade.apiToken` and other values relevant to the deploy included.  Or... leave as is :)  What do you think?

Reply:
I think _all_ our gateways have this problem ever since we made the improvement that we check for missing Brigade API token.

Personally, I usually modify the values.yaml that's included in the chart before ever invoking that target... buuuut... we've all seen me accidentally include a token in a commit before, so clearly that isn't really the best idea.

There's pretty much no getting around the fact that you need a legit API token here.

What do you think if we have the make target inject it using `--set` and an env var? We could actually check that the env var is set as the first bit of the make target. This would largely eliminate the danger of accidentally committing a token.

Reply:
If you like that, we should do it with all the gateways and other peripherals.

Reply:
I could go with the --set + env var option... but since there is usually further config needed for the gateway to be fully operational (well, definitely for this Slack gateway), would it make sense to go with a --file + env var representing the path to the values.yaml file?

Not sure what's best.  Since this affects other gateway deploy targets, it can of course be a follow-up.  Perhaps we create an issue generic to affected gateways...

Reply:
Took a stab at this. PTAL when it's convenient and LMK what you think.